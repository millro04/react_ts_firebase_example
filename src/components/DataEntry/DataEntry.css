.data-entry-wrapper {
    width: 100%;
    padding-top: 25px;
    display: flex;
    align-Items: center;
    justify-content: center;
}