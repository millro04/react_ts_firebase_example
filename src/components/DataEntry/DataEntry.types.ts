import { IBookData } from '../types';

export interface IDataEntryState {
    bookData: IBookData[],
    isLoading: boolean
  }