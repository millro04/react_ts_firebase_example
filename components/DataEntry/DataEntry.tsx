import React from 'react';
import DataEntryView from './DataEntryView';

export default class DataEntry extends React.Component {
  constructor(props: any) {
      super(props);
  }
  render() {
    return (
      <DataEntryView />
    );
  }
}