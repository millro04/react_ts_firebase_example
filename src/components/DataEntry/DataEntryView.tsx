import React from 'react';
import MaterialTable from 'material-table';
import './DataEntry.css';

interface IDataEntryViewProps {
  columns: any,
  tableData: any,
  onRowAdd: any,
  onRowUpdate: any,
  onRowDelete: any
}

export default class DataEntryView extends React.Component<IDataEntryViewProps, any> {
  constructor(props : IDataEntryViewProps) {
    super(props);
  }

  render() {
    return (
      <div className='data-entry-wrapper'>
        <MaterialTable
          style={{ width: '90%' }}
          columns={this.props.columns}
          data={this.props.tableData}
          title="Book Record"
          editable={{
            onRowAdd: this.props.onRowAdd,
            onRowUpdate: this.props.onRowUpdate,
            onRowDelete: this.props.onRowDelete
          }}
        />
      </div>
    );
  }
}