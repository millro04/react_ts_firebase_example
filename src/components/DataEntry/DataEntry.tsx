import React from 'react';
import DataEntryView from './DataEntryView';

const columns = [
  {
    title: 'Date',
    field: 'date'
  },
  {
    title: 'Book',
    field: 'book_name'
  },
  {
    title: 'Minutes Read',
    field: 'min_read'
  },
];

const tableData = [
  {
    date: '2020-02-01', book_name: 'Harry Potter 1', min_read: 60, id: 0
  },
  {
    date: '2020-02-01', book_name: 'Harry Potter 2', min_read: 30, id: 1
  }
];

interface IDataEntryState {
  bookData: any,
}

export default class DataEntry extends React.Component<any, IDataEntryState> {
  constructor(props: React.ReactPropTypes) {
      super(props);
      this.onRowAdd = this.onRowAdd.bind(this);
      this.onRowUpdate = this.onRowUpdate.bind(this);
      this.onRowDelete = this.onRowDelete.bind(this);
      this.state = {bookData: null};
  }

  componentDidMount() {
    this.getBookData();
  }

  getBookData() {
    this.setState({bookData: tableData});
  }

  onRowAdd(newData: any) {
    return new Promise((resolve, reject) => {

      try {
        const data = this.state.bookData;
        data.push(newData);
        this.setState({ bookData: data });
        resolve();
      }

      catch {
        reject();
      }
  });
  }

  onRowUpdate(newData: any, oldData: any) {
    return new Promise((resolve, reject) => {

      try {
        const data = this.state.bookData;
        data[newData.id] = newData;
        this.setState({ bookData: data });
        resolve();
      }

      catch {
        reject();
      }
  });
  }

  onRowDelete(oldData: any) {
    return new Promise((resolve, reject) => {

      try {
        let data = this.state.bookData;
        const index = data.indexOf(oldData);
        data.splice(index, 1);
        this.setState({ bookData: data });
        resolve();
      }

      catch {
        reject();
      }
  });
  }

  render() {
    return (
      <DataEntryView
        columns={columns}
        tableData={this.state.bookData}
        onRowAdd={this.onRowAdd}
        onRowUpdate={this.onRowUpdate}
        onRowDelete={this.onRowDelete}
        />
    );
  }
}