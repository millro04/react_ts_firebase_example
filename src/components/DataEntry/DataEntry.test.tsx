import React from 'react';
import DataEntry from './DataEntry';
import DataEntryView from './DataEntryView';
import { shallow, ShallowWrapper, configure, ReactWrapper, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({adapter: new Adapter()});


test('Data Entry component renders', () => {
    const component: ShallowWrapper = shallow(<DataEntry />);
    expect(component).toMatchSnapshot();
});


test('Data Entry Component renders its View Component', () => {
  const component: ShallowWrapper = shallow(<DataEntry />);
  expect(component.find(DataEntryView).length).toBeTruthy();
});


test('Data Entry View renders', () => {
    const onAdd = jest.fn();
    const onUpdate = jest.fn();
    const onDelete = jest.fn();

    const tableData = [
        {
            mock_field_value: 0
        }
    ]

    const mockColumns = [
        {
            title: 'Mock Title',
            field: 'mock_field_value'
        }
    ];
    const component: ShallowWrapper = shallow(
        <DataEntryView
            columns={mockColumns}
            tableData={tableData}
            onRowAdd={onAdd}
            onRowUpdate={onUpdate}
            onRowDelete={onDelete}
        />
    );
    expect(component).toMatchSnapshot();
  });

test('DataEntryView snapshot test', () => {
    const onAdd = jest.fn();
    const onUpdate = jest.fn();
    const onDelete = jest.fn();

    const tableData = [
        {
            mock_field_value: 0
        },
        {
            mock_field_value: 1
        },
        {
            mock_field_value: 2
        }
    ]

    const mockColumns = [
        {
            title: 'Mock Title',
            field: 'mock_field_value'
        }
    ];
    const component: ShallowWrapper = shallow(
        <DataEntryView
            columns={mockColumns}
            tableData={tableData}
            onRowAdd={onAdd}
            onRowUpdate={onUpdate}
            onRowDelete={onDelete}
        />
    );
    expect(component).toMatchSnapshot();
});
