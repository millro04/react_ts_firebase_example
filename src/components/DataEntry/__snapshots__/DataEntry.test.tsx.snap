// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Entry View renders 1`] = `ShallowWrapper {}`;

exports[`Data Entry component renders 1`] = `ShallowWrapper {}`;

exports[`DataEntryView snapshot test 1`] = `ShallowWrapper {}`;
